### PR TITLE
uart: various fixes & improvements

### PR DESCRIPTION
**Pull request by ghost**
* Add uart_get_actual_baudrate function
* Avoid comparing signed/unsigned integers
* Avoid using signed integers for indexing arrays
* Fix uart_release function

### CODE REVIEW

File: src/core/uart.c [hunk 1]
```diff
@@ -125,13 +125,34 @@ int uart_get_baudrate(uint32_t *baudrate)
 #endif
 }
 
+int uart_get_actual_baudrate(uint32_t *baudrate)
+{
+#ifndef __USE_UART_PORT3__
+    fprintf(stderr, "uart: __USE_UART_PORT3__ not defined\n");
+    return -1;
+#else
+    uint32_t x = 4;
```
Comment (mtusnio) [inline src/core/uart.c]: A small comment explaining what's x and why it's 4 or 16 would be useful.

**Review by mtusnio — CHANGES_REQUESTED** [review]

File: include/core/uart.h [hunk 2]
```diff
@@ -66,13 +66,28 @@ int uart_set_baudrate(uint32_t baudrate);
...
+ * @brief Get the actual speed of the current UART device.
+ *
+ * The device must be initialised first.
+ * This function differs from uart_get_baudrate since it computes the baudrate
+ * from UART3 registers. This gives the exact baudrate used by the UART device.
+ *
+ * @param[out] baudrate Actual baud rate of the UART device (must not be null)
+ * @return 0 if successful, -1 otherwise
+ */
+int uart_get_actual_baudrate(uint32_t *baudrate);
```
Comment (mtusnio) [inline include/core/uart.h]: Nitpicking here, but I'd rename this to "uart_get_real_baudrate" instead.

**Review by mtusnio — CHANGES_REQUESTED** [review]

File: include/core/uart.h (hunk 2 above)
Comment (ghost) [reply include/core/uart.h]: Done

File: src/core/uart.c (hunk 1 above)
Comment (ghost) [reply src/core/uart.c]: Done

**Review by mtusnio — APPROVED** [review]

**Pull request merged**